zookeeper-server/src/test/java/org/apache/zookeeper/ClientRequestTimeoutTest.java

```diff
@@ -19,78 +19,84 @@
 package org.apache.zookeeper;
 
 import static org.apache.zookeeper.test.ClientBase.CONNECTION_TIMEOUT;
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertTrue;
-import static org.junit.Assert.fail;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTimeout;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
 import java.io.IOException;
+import java.time.Duration;
 import org.apache.zookeeper.ZooDefs.Ids;
 import org.apache.zookeeper.client.HostProvider;
 import org.apache.zookeeper.client.ZKClientConfig;
 import org.apache.zookeeper.server.quorum.QuorumPeerTestBase;
 import org.apache.zookeeper.test.ClientBase;
 import org.apache.zookeeper.test.ClientBase.CountdownWatcher;
-import org.junit.Test;
+import org.junit.jupiter.api.Test;
 
 public class ClientRequestTimeoutTest extends QuorumPeerTestBase {
 
     private static final int SERVER_COUNT = 3;
     private boolean dropPacket = false;
     private int dropPacketType = ZooDefs.OpCode.create;
 
-    @Test(timeout = 120000)
+    @Test
```

Comment:
I see there is a @Timeout annotation since 5.5. Although it is experimental. Did you try it maybe?
https://junit.org/junit5/docs/5.5.0-RC2/api/org/junit/jupiter/api/Timeout.html

Reply:
assertTimeout is stable, so I preferred to use it.
But I can try it out.

Reply:
I would much prefer to use the annotation (for readability / clean code), if the timeout is really about the whole test and not only some small part of it. But only if it is stable. I wonder why is this experimental still. It was introduced in 5.5 and we are using 5.6 here.

Reply:
Hmm, tried it out with a few cases, looks working.

Reply:
it is a huge work to change it back to annotations for all test cases... let's wait for the opinion of others, before you would change it

Reply:
@symat I've uploaded a change to use Timeout annotation. It looks working for me.
This way we need much less code changes which helps backportability.